### ISSUE TITLE
About Meson build

### ISSUE THREAD

**Issue by Fenyutanchan** [JuliaPackaging/Yggdrasil#6488]
There are two problems with the Meson building system.

The first is 
```log
clang-13: error: no such file or directory: '/workspace/destdir/opt/aarch64-apple-darwin20/aarch64-apple-darwin20/sys-root/usr/local/lib/libgmp.dylib'
```
However, there is no `/workspace/destdir/opt/` in the building system.
I'd like to use `sed -i "s/\/workspace\/destdir\/opt/\/opt/g" build.ninja` to fix it.
Is it right?

The second is for the old version meson.
It seems that the old version meson could not resolve the darwin building.
I added `pip3 install -U meson` for upgrading the meson building system.
Maybe the meson building system should be upgraded to v1.0.1 in the BinaryBuilder.jl now?

**Comment by giordano** [comment]
> However, there is no /workspace/destdir/opt/ in the building system.
> I'd like to use sed -i "s/\/workspace\/destdir\/opt/\/opt/g" build.ninja to fix it. (see https://github.com/JuliaPackaging/Yggdrasil/pull/6485 for more details)
> Is it right?

Not really, it feels like there's something wrong in the build system of the specific package.
